simtools/data_model/meta_data_model.py

```diff
@@ -0,0 +1,307 @@
+"""
+Definition of metadata model for input to and output of gammasim-tools.
+Follows CTA top-level data model definition.
+Metadata is broken into hierarchical categories in this module.
+
+Metadata definitions require the three following fields to be defined:
+
+* type: data type (str)
+* required: boolean if this field is required as input(!)
+* default: default value
+
+These definitions are used for:
+
+* data products submitted to SimPipe ('input')
+* data products generated by SimPipe ('output')
+
+"""
+
+
+class InvalidSchemaList(Exception):
+    """
+    Exception raised for requests of unknown schema lists
+    """
+
+
+def _category_reference():
+    """
+    Metadata field REFERENCE.
+
+    Refers to the CTA Top-Level Data Model (document CTA-SPE-OSO-000000-0001.)
+
+    """
+    return {"VERSION": {"type": "str", "required": True, "default": "1.0.0"}}
+
+
+def _category_contact():
+    """
+    Metadata field CONTACT.
+
+    Organisation and person submitting or processing these data.
+
+    """
+    return {
+        "ORGANIZATION": {"type": "str", "required": True, "default": None},
+        "NAME": {"type": "str", "required": True, "default": None},
+        "EMAIL": {"type": "email", "required": True, "default": None},
+    }
+
+
+def _category_product():
+    """
+    Metadata field PRODUCT.
+
+    Describes the data product, especially its validity and associations.
+
+    """
+    return {
+        "DESCRIPTION": {"type": "str", "required": True, "default": None},
+        "CREATION_TIME": {"type": "datetime", "required": True, "default": None},
+        "ID": {"type": "str", "required": False, "default": None},
+        "DATA": {
+            "CATEGORY": {"type": "str", "required": False, "default": None},
+            "LEVEL": {"type": "str", "required": False, "default": None},
+            "TYPE": {"type": "str", "required": False, "default": None},
+            "MODEL": {
+                "NAME": {"type": "str", "required": False, "default": None},
+                "VERSION": {"type": "str", "required": False, "default": None},
+                "URL": {"type": "str", "required": False, "default": None},
```

Comment:
Could we add a `url` type like the email and instrumentlist ones?

Reply:
This is actually a rabbit hole and it seems really complicated to check for a valid url (I think it is an unsolved problem)?

See e.g., 
- https://stackoverflow.com/questions/7160737/how-to-validate-a-url-in-python-malformed-or-not
- https://stackoverflow.com/questions/827557/how-do-you-validate-a-url-with-a-regular-expression-in-python

It would be very nice to have, but...

Reply:
OK, no problem! 
(I saw those stackoverflow entries before asking and figured maybe we can do something simple, but it's not really worth it, you are right.)